### PR TITLE
feat(chart): Support an existing secret and configmap

### LINKED ISSUE #1257: Add support for existing secret and configmap in Helm chart
### Problem

The current helm chart does not support existing external secret and configmaps

### Solution

Add support for existing secret and configmap in Helm chart

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by josedev-union**
**Description**:
This PR is to support the existing secret or configmap for the relay configuration.
The typical usecase is when the sensitive data is stored in the vault software and the k8s secret is generated by external-secret operator.

**Related issue(s)**:

Fixes #1257 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by Nana-EC — APPROVED** [review]

File: helm-chart/templates/deployment.yaml [hunk 1]
```diff
@@ -38,7 +38,11 @@ spec:
         - configMapRef:
             name: {{ include "json-rpc-relay.fullname" . }}
         - secretRef:
+        {{- if .Values.existingSecret }}
+            name: {{ .Values.existingSecret }}
```
Comment (steven-sheehy) [inline helm-chart/templates/deployment.yaml:42]: DRY:

```suggestion
        {{- with .Values.existingSecret }}
            name: {{ . }}
```

Comment (steven-sheehy) [inline helm-chart/templates/deployment.yaml:42]: It should probably be templated so you can embed the release name in it to support multiple deploys in a single cluster.

File: helm-chart/templates/deployment.yaml [hunk 2]
```diff
@@ -38,7 +38,11 @@ spec:
...
+        {{- else }}
             name: {{ include "json-rpc-relay.fullname" . }}  
+        {{- end }}
```
Comment (steven-sheehy) [inline helm-chart/templates/deployment.yaml:45]: Is this really the best approach to take? We use a more simple, but customizable approach on the mirror node where we have a single [envFrom](https://github.com/hashgraph/hedera-mirror-node/blob/37a98829aa97872421435a3c7d6c92df603766de/charts/hedera-mirror-web3/values.yaml#L42) that users can add custom secrets or configmaps with templating support. So relay could do the same and have the existing secret and configmap moved to values.yaml and allow you to override it for your purposes.

**Review by Nana-EC — APPROVED** [review]

File: helm-chart/templates/deployment.yaml [hunk 3]
```diff
@@ -38,7 +38,10 @@ spec:
         - configMapRef:
             name: {{ include "json-rpc-relay.fullname" . }}
         - secretRef:
-            name: {{ include "json-rpc-relay.fullname" . }}  
+            name: {{ include "json-rpc-relay.fullname" . }}
+        {{- with .Values.extraEnvFrom }}
```
Comment (Nana-EC) [inline helm-chart/templates/deployment.yaml:42]: nit: envFrom would be simpler to conform with Mirror Node actually
```suggestion
        {{- with .Values.envFrom }}
```

Comment (Nana-EC) [inline helm-chart/templates/deployment.yaml:42]: Also not sure if you need the with but could also simplify with below. Again from mirror node
```suggestion
        envFrom: {{ tpl (toYaml .Values.envFrom) . | nindent 12 }}
```

**Review by Nana-EC — CHANGES_REQUESTED** [review]
LG, nit on naming to make the charts more consistent across the org

File: helm-chart/templates/deployment.yaml (hunk 3 above)
Comment (josedev-union) [reply helm-chart/templates/deployment.yaml:42]: Well, to make values identical across org charts as much as possible  is a good point. Just my concern is that `envFrom` is strictly used to replace the `envFrom` of a container spec, but we have 2 items for `envFrom` in default and users will just add more items using this value. So i think `extra` would be more reasonable and meaningful.
In other charts, we can also use `envFrom` and `extraEnvFrom` according to their manifest status.

Comment (Nana-EC) [reply helm-chart/templates/deployment.yaml:42]: Makes sense. Thanks for clarifying

Comment (Nana-EC) [reply helm-chart/templates/deployment.yaml:42]: Based on above I guess this also doesn't apply.

**Review by Nana-EC — APPROVED** [review]

**Pull request merged**